keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1083,6 +1105,38 @@ repeated evaluations if they are shared across validations.
     - expression: "itemMetadataNames.exists(name, name == 'required')"
 ```
 
+Since the variable names must be valid CEL names, this is checked at creation
+time.
+
+For resource constraint purposes, lazily-evaluated variables
+count towards the overall policy limit, but only once, to account for
+memoization.
+
+Informational type checking also applies to subexpressions.
+
+Subexpressions can reference other subexpressions, but circular references
+are not allowed. It is possible in theory to have two subexpressions that
+refer to each other that will never run at the same time, for instance:
+
+```
+#subExpr1
+someBool? "A" : subExpr2
+
+#subExpr2
+!someBool? subExpr1 : "B"
+```
+
+In practice, it is not clear what this would add, however. It could also
+lead to cases where a certain set of inputs causes a circular reference.
+So, it is safer to simply not allow circular references at all.
```

Comment:
Please decide.

FWIW, I'm in favor of variables being ordered and only being able to refer to other variables earlier in the list.

Reply:
Also, the decision here impacts how the variables need to be structured? Right now the example shows the variables defined as an ordered list, but if order does not matter I would have expected a map.

Reply:
I'll re-word the KEP to make it more clear I'm not in favor of circular references.

As long as there are no circular references I think an ordered list makes sense as well - I'll make that more clear as well.